### ISSUE TITLE
numpy: `--with-check` fails

### ISSUE THREAD

**Issue by sjackman** [Homebrew/homebrew-science#3863]
_From @AlecWong on July 25, 2016 8:22_

# Please follow the general troubleshooting steps first:
Yes I have tried the following tips:

-  Ran `brew update` and retried your prior step? 
-  Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
-  If you're seeing permission errors tried running `sudo chown -R $(whoami) $(brew --prefix)`?

### Bug reports:

```
==> python setup.py build --fcompiler=gnu95 --parallel=12 install --prefix=/home/username/.linuxbrew/Cellar/numpy/1.11
==> python -c import numpy; assert numpy.test().wasSuccessful()
Last 15 lines from /home/usrname/.cache/Homebrew/Logs/numpy/03.python:

\----------------------------------------------------------------------
Ran 6151 tests in 53.665s

FAILED (KNOWNFAIL=6, SKIP=11, errors=1)
Running unit tests for numpy
NumPy version 1.11.1
NumPy relaxed strides checking option: False
NumPy is installed in /home/username/.linuxbrew/Cellar/numpy/1.11.1/lib/python2.7/site-packages/numpy
Python version 2.7.12 (default, Jun 28 2016, 08:31:05) [GCC 6.1.1 20160602]
nose version 1.3.7
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AssertionError

READ THIS: https://github.com/Linuxbrew/brew/blob/master/share/doc/homebrew/Troubleshooting.md#troubleshooting
If reporting this issue please do so at (not Homebrew/brew):
  https://github.com/Homebrew/homebrew-python/issues
<hr>
```

_Copied from original issue: Linuxbrew/brew#84_

**Comment by dpo** [comment]
Shouldn't this go to homebrew/python?

**Comment by sjackman** [comment]
It should, but I haven't yet discussed with that tap whether they do or want to support Linuxbrew. Just assigning them a random ticket didn't seem to be the right way to go about it.

**Comment by dpo** [comment]
Sounds good.

**Comment by sjackman** [comment]
I also don't yet have a fix for this issue. Rather wish I did. A fair number of bioinformatics formula depend on `homebrew/python/numpy`.

**Comment by dpo** [comment]
Do you have the full logs that show what the error is exactly?

**Comment by sjackman** [comment]
Here's a failure on Travis: https://travis-ci.org/Linuxbrew/homebrew-science/jobs/145640307
It doesn't have the complete logs, but I can get those.

**Comment by dpo** [comment]
Yes, it's hard to tell anything from that Travis log.

**Comment by sjackman** [comment]
It's reproducible with Docker:
```
docker run -it linuxbrew/linuxbrew
brew install homebrew/python/numpy
```

`gist-logs`: https://gist.github.com/anonymous/e88c4f5311caca5c7dd6cd263d5f1c91

The relevant error is `relocation R_X86_64_PC32 against symbol `fun_' can not be used when making a shared object; recompile with -fPIC`

```
compiling Fortran sources
Fortran f77 compiler: /home/linuxbrew/.linuxbrew/bin/gfortran -Wall -g -ffixed-form -fno-second-underscore -Os -w -pipe -march=native -O3 -funroll-loops
Fortran f90 compiler: /home/linuxbrew/.linuxbrew/bin/gfortran -Wall -g -fno-second-underscore -Os -w -pipe -march=native -O3 -funroll-loops
Fortran fix compiler: /home/linuxbrew/.linuxbrew/bin/gfortran -Wall -g -ffixed-form -fno-second-underscore -Wall -g -fno-second-underscore -Os -w -pipe -march=native -O3 -funroll-loops
creating /tmp/tmpBK1aR0/tmp/tmpAAAOIy
compile options: '-I/tmp/tmpBK1aR0/src.linux-x86_64-2.7 -I/home/linuxbrew/.linuxbrew/Cellar/numpy/1.11.1/lib/python2.7/site-packages/numpy/core/include -I/home/linuxbrew/.linuxbrew/include -I/home/linuxbrew/.linuxbrew/opt/openssl/include -I/home/linuxbrew/.linuxbrew/opt/sqlite/include -I/home/linuxbrew/.linuxbrew/Cellar/python/2.7.12/include/python2.7 -c'
gfortran:f77: /tmp/tmpAAAOIy/tmpPNC0ZM.f
/home/linuxbrew/.linuxbrew/bin/gfortran -Wall -g -L/home/linuxbrew/.linuxbrew/lib -Wl,--dynamic-linker=/home/linuxbrew/.linuxbrew/lib/ld.so -Wl,-rpath,/home/linuxbrew/.linuxbrew/lib -shared /tmp/tmpBK1aR0/tmp/tmpBK1aR0/src.linux-x86_64-2.7/_test_ext_module_5403module.o /tmp/tmpBK1aR0/tmp/tmpBK1aR0/src.linux-x86_64-2.7/fortranobject.o /tmp/tmpBK1aR0/tmp/tmpAAAOIy/tmpPNC0ZM.o -L/home/linuxbrew/.linuxbrew/lib -L/home/linuxbrew/.linuxbrew/opt/openssl/lib -L/home/linuxbrew/.linuxbrew/opt/sqlite/lib -L/home/linuxbrew/.linuxbrew/Cellar/python/2.7.12/lib -lpython2.7 -lgfortran -o ./_test_ext_module_5403.so
/usr/bin/ld: /tmp/tmpBK1aR0/tmp/tmpAAAOIy/tmpPNC0ZM.o: relocation R_X86_64_PC32 against symbol `fun_' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
/usr/bin/ld: /tmp/tmpBK1aR0/tmp/tmpAAAOIy/tmpPNC0ZM.o: relocation R_X86_64_PC32 against symbol `fun_' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
error: Command "/home/linuxbrew/.linuxbrew/bin/gfortran -Wall -g -L/home/linuxbrew/.linuxbrew/lib -Wl,--dynamic-linker=/home/linuxbrew/.linuxbrew/lib/ld.so -Wl,-rpath,/home/linuxbrew/.linuxbrew/lib -shared /tmp/tmpBK1aR0/tmp/tmpBK1aR0/src.linux-x86_64-2.7/_test_ext_module_5403module.o /tmp/tmpBK1aR0/tmp/tmpBK1aR0/src.linux-x86_64-2.7/fortranobject.o /tmp/tmpBK1aR0/tmp/tmpAAAOIy/tmpPNC0ZM.o -L/home/linuxbrew/.linuxbrew/lib -L/home/linuxbrew/.linuxbrew/opt/openssl/lib -L/home/linuxbrew/.linuxbrew/opt/sqlite/lib -L/home/linuxbrew/.linuxbrew/Cellar/python/2.7.12/lib -lpython2.7 -lgfortran -o ./_test_ext_module_5403.so" failed with exit status 1
```
https://gist.github.com/anonymous/e88c4f5311caca5c7dd6cd263d5f1c91#file-03-python-L105

**Comment by dpo** [comment]
Perhaps the reason is that one of the three .o's (_test_ext_module_5403module.o, fortranobject.o or tmpPNC0ZM.o) wasn't compiled with -fPIC. But I'm not sure how to force that flag.

**Comment by sjackman** [comment]
I believe this build failure is an upstream issue. I opened a ticket upstream: https://github.com/numpy/numpy/issues/7794

**Comment by ilovezfs** [comment]
@sjackman I believe this has been going on since at least https://github.com/Homebrew/homebrew-python/pull/294 CC @tdsmith

**Comment by sjackman** [comment]
https://github.com/Homebrew/homebrew-python/pull/294#issuecomment-204658192
> The fortran test error is weird but doesn't happen for me locally so I deem this Probably Okay.

The fortran test error prevents `homebrew/python/numpy` from building on Linux. Any hope of fixing the test failure? Can the test be disabled if it's safe to ignore it?

**Comment by ilovezfs** [comment]
Yes, I think we should disable it for now. Just unclear if it's the same thing as what happened with 1.11.0 since you were saying you think this is new to 1.11.1?

**Comment by sjackman** [comment]
I have these versions of `numpy` installed on my system:
`1.9.2_1 1.9.3 1.10.0 1.10.1 1.10.4 1.11.0`
I'm unable to build `1.11.1`.

**Comment by ilovezfs** [comment]
```
==> Summary
🍺  /usr/local/Cellar/numpy/1.11.1: 1,560 files, 31.9M, built in 3 minutes 21 seconds
iMac-TMP:~ joe$ brew test -v numpy
Testing homebrew/python/numpy
==> Using the sandbox
/usr/bin/sandbox-exec -f /tmp/homebrew20160729-48657-1b8nnir.sb /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby -W0 -I /usr/local/Library/Homebrew -- /usr/local/Library/Homebrew/test.rb /usr/local/Library/Taps/homebrew/homebrew-python/numpy.rb -v
==> python -c import numpy as np
t = np.ones((3,3), int)
assert t.sum() == 9
assert np.dot(t, t).sum() == 27
```

Try --with-python3

**Comment by sjackman** [comment]
I am using `--with-python3` but on Linux.

**Comment by ilovezfs** [comment]
Is it actually using python3? Maybe try making it "python3" instead of ":python3"?

**Comment by ilovezfs** [comment]
@sjackman I think we need to add `"without-python"` to the numpy options in addition to the `with-python3` that is there already.

**Comment by ilovezfs** [comment]
@sjackman I think this may be the problem:
https://github.com/numpy/numpy/issues/6809
https://github.com/numpy/numpy/issues/1171

**Comment by sjackman** [comment]
Aha! Smoking gun. Yes, that looks promising. Although it seems upstream thinks it's a `wontfix` bug. That's not promising.

**Comment by ilovezfs** [comment]
Yup! Is it reproducible outside of brew? Not sure if it's that we're setting `-march=foo` or flag refurbishing or some other `magical_voodoo.rb` where we end up stepping on upstream's flags, if that is indeed what's going on.

**Comment by sjackman** [comment]
Completely unconfirmed speculation follows. Based on the title of #6809, it looks as though upstream sets `-fPIC` in `FFLAGS` (which is not the correct place to set it) and Brew's overriding `FFLAGS` causes the shared objects to be compiled without `-fPIC` and then linking fails.

**Comment by tdsmith** [comment]
Feel free to open issues on homebrew-python, though you may hear "patches welcome!" from me. If setting `-fPIC` somewhere fixes this I'm happy.

**Comment by sjackman** [comment]
If my reading of the above is correct then `ENV.append "FFLAGS", "-fPIC"` may resolve the issue.

**Comment by tdsmith** [comment]
Fixed by Homebrew/homebrew-python@4c5a6fe9.

**Comment by sjackman** [comment]
Thanks, Tim!
